Documentation/ApiOverview/Authentication/CSRFlikeRequestTokenHandling.rst

```diff
@@ -0,0 +1,163 @@
+..  include:: /Includes.rst.txt
+..  index:: Authentication, Request token
+..  _authentication-request-token:
+
+================================
+CSRF-like request token handling
+================================
+
+..  versionadded:: 12.0
+
+A CSRF-like request token handling is available to mitigate potential cross-site
+requests on actions with side effects. This approach does not require an
+existing server-side user session, but uses a nonce (number used once) as a
+"pre-session". The main scope is to ensure a user actually has visited a page,
+before submitting data to the webserver.
+
+This token can only be used for HTTP methods `POST`, `PUT` or `PATCH`, but
+for instance not for `GET` request.
+
+The :php:`\TYPO3\CMS\Core\Middleware\RequestTokenMiddleware` resolves
+request tokens and nonce values from a request and enhances responses with
+a nonce value in case the underlying application issues one. Both items are
+serialized as JSON Web Token (JWT) hash signed with `HS256`. Request tokens
+use the provided nonce value during signing.
+
+Session cookie names involved for providing the nonce value:
+
+*   `typo3nonce_[hash]` in case request served with plain HTTP
+*   `__Secure-typo3nonce_[hash]` in case request served with secured HTTPS
+
+Submitting request token value to application:
+
+*   HTTP body, e.g. in `<form>` via parameter `__RequestToken`
+*   HTTP header, e.g. in XHR via header `X-TYPO3-Request-Token`
+
+..  attention::
+    When working with multiple browser tabs, an existing nonce value (stored as
+    session cookie in the browser of the user) might be overridden.
+
+..  note::
+    The current concept uses the :php:`\TYPO3\CMS\Core\Security\NoncePool` which
```

Comment:
link this to github?

Reply:
I see no value for the user (also this class is marked as internal). It is interesting to look into it from a "how is this done" view.